lib/puppet/transport/cisco_ios.rb

```diff
@@ -0,0 +1,370 @@
+      # eg. using --wait
+      dirname = File.dirname(known_hosts_file)
+      unless File.directory?(dirname)
+        FileUtils.mkdir_p(dirname)
+      end
+
+      verify_host_key = (Gem.loaded_specs['net-ssh'].version < Gem::Version.create('4.2.0')) ? :paranoid : :verify_host_key
+      session = if !config[:verify_hosts].nil? && !config[:verify_hosts]
+                  Net::SSH.start(config[:host],
+                                 config[:user],
+                                 password: config[:password].unwrap,
+                                 port: config[:port] || 22,
+                                 timeout: config[:timeout] || 30,
+                                 verify_host_key => false,
+                                 user_known_hosts_file: known_hosts_file)
+                else
+                  Net::SSH.start(config[:host],
+                                 config[:user],
+                                 password: config[:password].unwrap,
+                                 port: config[:port] || 22,
+                                 timeout: config[:timeout] || 30,
+                                 verify_host_key => :very,
+                                 user_known_hosts_file: known_hosts_file)
+                end
+
+      @options = { 'Prompt' =>  %r{#{commands['default']['connect_prompt']}},
+                   'Session' => session }
+
+      if config[:ssh_logging] == true && (debug || Puppet::Util::Log.level == :debug)
+        if config[:ssh_log_file]
+          @options['Dump_log'] = config[:ssh_log_file]
+        else
+          # ensure we have a cache folder structure exists for the device
+          FileUtils.mkdir_p(Puppet[:statedir]) unless File.directory?(Puppet[:statedir])
+          @options['Dump_log'] = "#{Puppet[:statedir]}/SSH_I_DUMPED"
+        end
+        FileUtils.touch @options['Dump_log']
+        FileUtils.chmod 0o0640, @options['Dump_log']
+      end
+      @connection = Net::SSH::Telnet.new(@options)
+      @enable_password = config[:enable_password].unwrap
+      @command_timeout = config[:command_timeout] || 120
+      # IOS will page large results which breaks prompt search
+      send_command(@connection, 'terminal length 0')
+      @connection
+    end
+
+    def facts(_context)
+      facts = { 'operatingsystem' => 'cisco_ios' }
+      return_facts = {}
+      # https://www.cisco.com/c/en/us/support/docs/switches/catalyst-6500-series-switches/41361-serial-41361.html
+      begin
+        version_info = @connection.cmd('show version')
+        raise Puppet::Error, 'Could not retrieve facts' unless version_info
+        facts['hardwaremodel'] = version_info[%r{cisco\s+(\S+).+processor}i, 1]
+        facts['hostname'] = version_info[%r{(\S+)\s+uptime}, 1]
+        facts['serialnumber'] = version_info[%r{Processor board ID (\w*)}, 1]
+        facts['operatingsystemrelease'] = version_info[%r{(?i)IOS Software.*Version\s+([^,\s]+)}, 1]
+      end
+      return_facts.merge(facts)
```

Comment:
`return_facts` is just an empty hash, can be removed and `facts` used for everything